### ISSUE TITLE
Add pixel-wise options to grasp model, Integrate Spatial Softmax

### ISSUE THREAD

**Issue by ahundt** [jhu-lcsr/costar_plan#295]
Add pixel-wise training options to grasp_train and grasp_model, incorporating spatial softmax with the option to not find argmax during training, so that the image coordinate defined by the actual robot ending pose can be directly accessed for backpropagation.
